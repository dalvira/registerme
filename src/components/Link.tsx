import React from 'react';
import {Text, StyleSheet} from 'react-native';
import {colors} from '../theme/colors';

interface Props {
  text: string;
  onPress: () => void;
}

const Link = (props: Props) => {
  return (
    <Text style={styles.text} onPress={props.onPress}>
      {props.text}
    </Text>
  );
};

export default Link;

const styles = StyleSheet.create({
  text: {
    color: colors.blue,
    fontSize: 16,
    fontWeight: '600',
  },
});
