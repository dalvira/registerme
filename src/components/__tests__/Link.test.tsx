import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link';

const mockCall = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(<Link text="Sign Up" onPress={mockCall} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
