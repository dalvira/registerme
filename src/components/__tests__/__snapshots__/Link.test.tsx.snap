// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  onPress={[MockFunction]}
  style={
    Object {
      "color": "#183bdb",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
>
  Sign Up
</Text>
`;
